egs2/TEMPLATE/st2/st2.sh

```diff
@@ -0,0 +1,2066 @@
+src_bpe_char_cover=1.0  # character coverage when modeling BPE for source language
+tgt_token_type=bpe      # Tokenization type (char or bpe) for target language.
+tgt_nbpe=30             # The number of BPE vocabulary for target language.
+tgt_bpemode=unigram     # Mode of BPE (unigram or bpe) for target language.
+tgt_bpe_input_sentence_size=100000000 # Size of input sentence for BPE for target language.
+tgt_bpe_nlsyms=         # non-linguistic symbols list, separated by a comma, for BPE for target language.
+tgt_bpe_char_cover=1.0  # character coverage when modeling BPE for target language.
+
+# Ngram model related
+use_ngram=false
+ngram_exp=
+ngram_num=3
+
+# Language model related
+use_lm=true       # Use language model for ST decoding.
+lm_tag=           # Suffix to the result dir for language model training.
+lm_exp=           # Specify the directory path for LM experiment.
+                  # If this option is specified, lm_tag is ignored.
+lm_stats_dir=     # Specify the directory path for LM statistics.
+lm_config=        # Config for language model training.
+lm_args=          # Arguments for language model training, e.g., "--max_epoch 10".
+                  # Note that it will overwrite args in lm config.
+use_word_lm=false # Whether to use word language model.
+num_splits_lm=1   # Number of splitting for lm corpus.
+# shellcheck disable=SC2034
+word_vocab_size=10000 # Size of word vocabulary.
+
+# ST model related
+st_tag=        # Suffix to the result dir for st model training.
+st_exp=        # Specify the directory path for ST experiment.
+                # If this option is specified, st_tag is ignored.
+st_stats_dir=  # Specify the directory path for ST statistics.
+st_config=     # Config for st model training.
+st_args=       # Arguments for st model training, e.g., "--max_epoch 10".
+                # Note that it will overwrite args in st config.
+ignore_init_mismatch=false      # Ignore initial mismatch
+num_splits_st=1                 # Number of splitting for lm corpus.
+speech_token_case="ts"          # speech discrete token  case. ts: true sequence, rm: remove duplicated tokens
+speech_token_lang="wavlm_large_21_km2000"  # speech discrete token type abbrev. id (e.g., wavlm_large_21_km2000)
+src_tgt_text_case="lc.rm"       # source / target transcript case. Note, all source / target text should use the same case for now.
+src_tgt_text_lang=en            # source / target language abbrev. id (e.g., en). Multiple langs are supported to support multiple tasks, with space between (e.g., "es/en"), from data's perspect of view, src_lang of text is the first.multiple tasks, with space between (e.g., "es/en"), from data's perspect of view, src_lang of text is the first.
+tgt_tasks="asr/st"              # task abbrev. id (e.g., st). Multiple tasks are supported to support multiple tasks, with space between (e.g., "asr/st")
```

Comment:
space between or "/" between?

Reply:
Since it was for multi-task, is it a good idea to put in st task or a more general one?

I'm asking cause the major reason we split st from asr in the previous context is:
- the different architectures between each other (for st only tasks, we also have a few unique architecture designs for each component (e.g., separate asr/mt decoder, two-pass framework with multi-decoder, etc.)
- data preparation (design for tgt_text, src_text, and src_speech)
- specific evaluation (bleu calculation and multi-reference support)

however, I feel many of the above parts are not shared (e.g., the architecture is still discrete asr; we skip the joint-task framework, instead, we do either asr or st in a single run; no support to multi-reference scoring)

Given the above reasons, I lean to call it s2t2 instead of st2. Please let me know you thoughts!